tests/mu_eat_empty_args.c

```diff
@@ -0,0 +1,200 @@
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "mu_eat_empty_args.h"
+
+/*this is both a "does it compile"? and a "does it produce the right values?" test*/
+/*this file wants to test just the MU_EAT_EMPTY_ARGS feature*/
+
+#define EMPTY_MACRO 
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS()
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(,)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, )
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(, EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, EMPTY_MACRO )
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, EMPTY_MACRO, EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, EMPTY_MACRO, )
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, , EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(, EMPTY_MACRO, EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(, , EMPTY_MACRO)
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(, EMPTY_MACRO, )
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(EMPTY_MACRO, , )
+
+/*the below line is expected to stop the compiler if it expands to anything... sort of*/
+MU_EAT_EMPTY_ARGS(, , )
+
+
+#define POOR_MANS_ASSERT_AND_RETURN(condition)                                                          \
+do                                                                                                      \
+{                                                                                                       \
+    if (!(condition))                                                                                   \
+    {                                                                                                   \
+        printf("condition \"" #condition "\" in LINE=" MU_TOSTRING(__LINE__) " was UNEXPECTEDLY FALSE\n");          \
+        return MU_FAILURE;                                                                              \
+    }                                                                                                   \
+} while (0)
+
+
+int run_mu_eat_empty_args_test(void)
```

Comment:
>run_mu_eat_empty_args_test [](start = 4, length = 26)

Cheers. This was very useful to just test my alternative implementation.

Reply:
Cheers!

---
In reply to: [270709079](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270709079) [](ancestors = 270709079)